### PR TITLE
Rebrand Classic as Reader mode and Paired as Transitional mode; add Exit Reader Mode link

### LINKED ISSUE #1943: Re-brand Classic mode as Reader mode
The Classic mode isn't going anywhere, but we don't want to encourage users to use it unless they explicitly understand that it provides a stripped-down basic template. (At least it allows their potentially-heavy site content to still be consumed on bandwidth-constrained devices.) To this end, we can re-purpose the Classic mode as if it is a mobile “Reader” mode similar to what some browsers offer today. The key difference about the Reader mode offered by the plugin, however, is that ads would still be possible (via `amp-ad`).

We may need to make it clear that this reader mode would not be presented to desktop visitors. Only visitors from services that surface AMP content would present users with this mode.

As a first step, the “Leave Comment” link in the Classic template could be just renamed to “Exit Reader Mode”.

### LINKED ISSUE #2022: Re-brand Paired mode as Transitional mode
The paired mode is actually a transitional mode aimed at helping developers traverse the path from no AMP to native AMP in an easier way. 

This issue addresses this by rebranding the `Paired Mode` to `Transitional Mode`.

/cc @westonruter

### PR DESCRIPTION
**Pull request by felixarntz**
Fixes #1943. Fixes #2022.

* Replaces Classic mode with Reader mode in wording.
* Adjusts relevant user-facing language to better match Reader mode intentions.
* Renames Paired mode to Transitional mode. 
* Updates description of the modes.
* Adds a link to the top-right corner of the Reader mode template which points to the same resource in non-AMP version.
* Fixes erroneous inclusion of AMP panel in Customizer when theme support is present (these settings only apply to Reader mode).

I decided on going with an additional link to the canonical version because using the current comments link for that would be misleading. It points to the comments section of the post, where what we want is just a link to exit Reader mode and view the canonical version. Furthermore, if we just replaced that "Leave a Comment" link with an "Exit Reader Mode" link, it would be an unexpected location for such a link, as you typically add comments below a post.

### CODE REVIEW

File: includes/options/class-amp-options-menu.php [hunk 1]
```diff
@@ -186,11 +186,11 @@ public function render_theme_support() {
 					<dt>
 						<input type="radio" id="theme_support_disabled" name="<?php echo esc_attr( AMP_Options_Manager::OPTION_NAME . '[theme_support]' ); ?>" value="disabled" <?php checked( $theme_support, 'disabled' ); ?>>
 						<label for="theme_support_disabled">
-							<strong><?php esc_html_e( 'Classic', 'amp' ); ?></strong>
+							<strong><?php esc_html_e( 'Reader', 'amp' ); ?></strong>
 						</label>
 					</dt>
 					<dd>
-						<?php esc_html_e( 'Display AMP responses in classic (legacy) post templates in a basic design that does not match your theme\'s templates.', 'amp' ); ?>
+						<?php esc_html_e( 'Display AMP responses in a simplified reader mode design that does not match your theme\'s templates.', 'amp' ); ?>
```
Comment (felixarntz) [inline includes/options/class-amp-options-menu.php]: We might need to discuss how to best phrase this here. Do we want to highlight that it's not respecting the theme's design? Should we maybe phrase this as "separate from your theme" to sound less negative?

File: includes/templates/class-amp-post-template.php [hunk 2]
```diff
@@ -267,23 +267,25 @@ private function build_post_data() {
...
-				'post_modified_timestamp' => $post_modified_timestamp,
-				'post_author'             => $post_author,
+				'post'                     => $this->post,
+				'post_id'                  => $this->ID,
+				'post_title'               => $post_title,
+				'post_publish_timestamp'   => $post_publish_timestamp,
+				'post_modified_timestamp'  => $post_modified_timestamp,
+				'post_author'              => $post_author,
+				'post_canonical_link_url'  => get_permalink( $this->ID ),
+				'post_canonical_link_text' => __( 'Exit Reader Mode', 'amp' ),
```
Comment (felixarntz) [inline includes/templates/class-amp-post-template.php]: Alternatively, we could say "View Canonical". What do you think?

File: readme.txt [hunk 3]
```diff
@@ -40,15 +40,15 @@ If you are a developer, we encourage you to [follow along](https://github.com/am
...
-1. The editor will surface validation issues during content authoring. The specific blocks with validation errors are indicated.
-1. Validated URLs include the list of validation errors encountered, giving control over whether sanitization for a validation error is accepted or rejected.
-1. Styles added by themes and plugins are automatically concatenated, minified, and tree-shaken to try to keep the total under 50KB of inline CSS.
-1. A WP-CLI command is provided to check the URLs on a site for AMP validity. Results are available in the admin for inspection.
+2. Many themes can be served as AMP without any changes; the default experience is as if JavaScript is turned off in the browser since scripts are removed.
+3. Reader mode templates are still available, but they are are limited. Not only do they differ from the active theme, any validation errors are silently sanitized.
+4. Switch from Reader mode to Paired or Native mode in AMP settings screen. You may need to disable the admin bar in AMP if your theme has a larger amount of CSS.
```
Comment (felixarntz) [inline readme.txt]: Screenshots 3 and 4 need to be refreshed. @westonruter @ThierryA Do you have the originals there?

File: includes/options/class-amp-options-menu.php (hunk 1 above)
Comment (swissspidy) [reply includes/options/class-amp-options-menu.php]: `Display AMP responses in a simplified reader mode design separate from your theme` could work

File: includes/templates/class-amp-post-template.php (hunk 2 above)
Comment (swissspidy) [reply includes/templates/class-amp-post-template.php]: I don't think the average user knows what "Canonical" means in this context. Not to speak of translators. "Original" or "Full Version" would work better though

`Exit Reader Mode` sounds reasonable though.

For comparison, Safari uses "Hide Reader View".

File: readme.md [hunk 4]
```diff
@@ -4,15 +4,15 @@
 ![Banner](wp-assets/banner-1544x500.png)
 Enable AMP on your WordPress site, the WordPress way.
 
-**Contributors:** [automattic](https://profiles.wordpress.org/automattic), [xwp](https://profiles.wordpress.org/xwp), [google](https://profiles.wordpress.org/google), [westonruter](https://profiles.wordpress.org/westonruter), [ryankienstra](https://profiles.wordpress.org/ryankienstra), [batmoo](https://profiles.wordpress.org/batmoo), [stubgo](https://profiles.wordpress.org/stubgo), [albertomedina](https://profiles.wordpress.org/albertomedina), [tweetythierry](https://profiles.wordpress.org/tweetythierry), [joshuawold](https://profiles.wordpress.org/joshuawold), [postphotos](https://profiles.wordpress.org/postphotos)  
-**Tags:** [amp](https://wordpress.org/plugins/tags/amp), [mobile](https://wordpress.org/plugins/tags/mobile)  
-**Requires at least:** 4.9  
-**Tested up to:** 5.1  
-**Stable tag:** 1.0.2  
-**License:** [GPLv2 or later](http://www.gnu.org/licenses/gpl-2.0.html)  
-**Requires PHP:** 5.4
```
Comment (swissspidy) [inline readme.md]: You should probably fix your IDE configuration to not strip whitespace at the end of the line. This is required here for Markdown and thus part of the `.editorconfig`: https://github.com/ampproject/amp-wp/blob/cd31949d1b49c0e2eae42c2c1efc15a0c1cf9862/.editorconfig#L17-L18

**Comment by felixarntz** [comment]
I will continue work on this on Monday, and also cover the rebranding of Paired mode, per our earlier conversation.

File: includes/templates/class-amp-post-template.php (hunk 2 above)
Comment (amedina) [reply includes/templates/class-amp-post-template.php]: I see the point of "View Canonical", but it is true that is will be confusing for most users. OTOH, having "Exit Reader Mode" implies that the user is aware that she is in Reader Mode. Probably we can express the duality by a exit route with a message along these lines: "You are viewing the Reader Mode version of `<this post>`", where `<this post>` links to the canonical version.

File: includes/options/class-amp-options-menu.php (hunk 1 above)
Comment (amedina) [reply includes/options/class-amp-options-menu.php]: Another alternative: `Generates AMP content using simplified templates, which are light but may not match the look-and-feel of your site`

**Comment by felixarntz** [comment]
The latest commits take care of rebranding Paired Mode and making the Reader mode exit link opt-in via a Customizer control.

**Comment by amedina** [comment]
Another piece of this PR: rebranding of Native mode to Strict AMP Mode.

**Comment by amedina** [comment]
I can't see the `Exit Reader Mode` link anymore, or the configuration option.

**Comment by felixarntz** [comment]
> I can't see the `Exit Reader Mode` link anymore, or the configuration option.

The exit link is now opt-in as you suggested. You can control it in the Customizer together with the other header bar customizations.

File: includes/options/class-amp-options-menu.php [hunk 5]
```diff
@@ -138,9 +138,11 @@ public function add_menu_items() {
...
-		$theme_support      = AMP_Options_Manager::get_option( 'theme_support' );
-		$native_description = __( 'Reuses active theme\'s templates to display AMP responses but does not use separate URLs for AMP. This means your site is <b>AMP-first</b> and your canonical URLs are AMP.', 'amp' );
-		$paired_description = __( 'Reuses active theme\'s templates to display AMP responses, but uses separate URLs for AMP. Each canonical URL may have a corresponding AMP URL, if the content is fully AMP valid.', 'amp' );
+		$theme_support = AMP_Options_Manager::get_option( 'theme_support' );
+
+		$native_description       = __( 'Reuses active theme\'s templates to display AMP responses but does not use separate URLs for AMP. This means your site is <b>AMP-first</b> and your canonical URLs are AMP.', 'amp' );
+		$transitional_description = __( 'Reuses active theme\'s templates to display AMP responses, but uses separate URLs for AMP. Each canonical URL may have a corresponding AMP URL, if the content is fully AMP valid.', 'amp' );
+		$reader_description       = __( 'Generates AMP content using simplified templates, which are light but may not match the look-and-feel of your site.', 'amp' );
```
Comment (westonruter) [inline includes/options/class-amp-options-menu.php]: I think this description should perhaps note something about classic, like “These include the classic templates.”

This also should note that only posts/pages can be served as AMP in this mode. It is not suitable for other kinds of queries.

File: includes/options/class-amp-options-menu.php [hunk 6]
```diff
@@ -138,9 +138,11 @@ public function add_menu_items() {
 	 * @since 1.0
 	 */
 	public function render_theme_support() {
-		$theme_support      = AMP_Options_Manager::get_option( 'theme_support' );
-		$native_description = __( 'Reuses active theme\'s templates to display AMP responses but does not use separate URLs for AMP. This means your site is <b>AMP-first</b> and your canonical URLs are AMP.', 'amp' );
-		$paired_description = __( 'Reuses active theme\'s templates to display AMP responses, but uses separate URLs for AMP. Each canonical URL may have a corresponding AMP URL, if the content is fully AMP valid.', 'amp' );
+		$theme_support = AMP_Options_Manager::get_option( 'theme_support' );
+
+		$native_description       = __( 'Reuses active theme\'s templates to display AMP responses but does not use separate URLs for AMP. This means your site is <b>AMP-first</b> and your canonical URLs are AMP.', 'amp' );
+		$transitional_description = __( 'Reuses active theme\'s templates to display AMP responses, but uses separate URLs for AMP. Each canonical URL may have a corresponding AMP URL, if the content is fully AMP valid.', 'amp' );
```
Comment (westonruter) [inline includes/options/class-amp-options-menu.php]: I think the transitional mode and (especially) reader mode should both include a note that says mobile visitors to origin will not be automatically redirected to the AMP version. The AMP version is presented to mobile users encountering your content via Google Search, Twitter, etc.

File: readme.txt [hunk 7]
```diff
@@ -28,8 +28,8 @@ With the official AMP plugin for WordPress, the WordPress ecosystem is provided
 == Installation ==
 
 1. Upload the folder to the `/wp-content/plugins/` directory.
-1. Activate the plugin through the 'Plugins' menu in WordPress.
-1. If you currently use older versions of the plugin in `Classic mode`, it is strongly encouraged to migrate to `Paired` or `Native mode`.
+2. Activate the plugin through the 'Plugins' menu in WordPress.
+3. If you currently use older versions of the plugin in `Reader` mode, it is strongly encouraged to migrate to `Transitional` or `Native` mode.
```
Comment (westonruter) [inline readme.txt]: The note about the behavior of reader/transitional modes should also be added here, that they are for mobile visitors from Google Search, Twitter, etc. No redirection is done. This is not a mobile theme.

/cc @amedina

File: readme.txt (hunk 3 above)
Comment (ThierryA) [reply readme.txt]: @felixarntz what to do you mean by originals?

Comment (westonruter) [reply readme.txt]: Original screenshots? Yes. For screenshot 3 I just went to https://2017-theme.amp-wp.org/roads-of-the-city/ and enabled classic mode.

And I took the screenshot of the AMP settings screen from that environment too.

**Comment by westonruter** [comment]
Updates to description after iterating with @amedina:

image

Outstanding is the styling of the exit link.

**Comment by westonruter** [comment]
When site icon is absent:

image

When site icon is present:

Screen Shot 2019-04-05 at 15 32 40

**Comment by amedina** [comment]
I like the design of the "Exit Reader Mode" style.

File: readme.txt (hunk 7 above)
Comment (amedina) [reply readme.txt]: Agree. Same line from the Admin Settings page would work.

**Pull request merged**